### ISSUE TITLE
fix compilation warnings for VS

### ISSUE THREAD

**Issue by StrikerRUS** [microsoft/LightGBM#2774]
Refer to https://lightgbm-ci.visualstudio.com/lightgbm-ci/_build/results?buildId=4710&view=logs&j=e0c3ed1e-0c79-508f-2647-33e12d6ed3d4&t=d764710f-c78e-54fd-dda4-11d6a80fefc7.
```
2020-02-19T01:27:27.9462816Z Mode                LastWriteTime         Length Name                                                                  
2020-02-19T01:27:27.9464751Z ----                -------------         ------ ----                                                                  
2020-02-19T01:27:27.9467953Z d-----        2/19/2020   1:27 AM                build                                                                 
2020-02-19T01:27:28.9117147Z -- Building for: Visual Studio 15 2017
2020-02-19T01:27:44.8797542Z -- The C compiler identification is MSVC 19.16.27035.0
2020-02-19T01:27:46.6904289Z -- The CXX compiler identification is MSVC 19.16.27035.0
2020-02-19T01:27:46.8190716Z -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
2020-02-19T01:27:48.3960092Z -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
2020-02-19T01:27:48.3994931Z -- Detecting C compiler ABI info
2020-02-19T01:27:49.3323508Z -- Detecting C compiler ABI info - done
2020-02-19T01:27:49.3356567Z -- Detecting C compile features
2020-02-19T01:27:49.3800533Z -- Detecting C compile features - done
2020-02-19T01:27:49.3985556Z -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
2020-02-19T01:27:50.2654801Z -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
2020-02-19T01:27:50.2674157Z -- Detecting CXX compiler ABI info
2020-02-19T01:27:51.1310360Z -- Detecting CXX compiler ABI info - done
2020-02-19T01:27:51.1340473Z -- Detecting CXX compile features
2020-02-19T01:27:51.1351734Z -- Detecting CXX compile features - done
2020-02-19T01:27:54.3455253Z -- Found OpenMP_C: -openmp (found version "2.0") 
2020-02-19T01:27:55.2324688Z -- Found OpenMP_CXX: -openmp (found version "2.0") 
2020-02-19T01:27:55.2336042Z -- Found OpenMP: TRUE (found version "2.0")  
2020-02-19T01:27:55.2654038Z -- Performing Test MM_PREFETCH
2020-02-19T01:27:55.8850870Z -- Performing Test MM_PREFETCH - Failed
2020-02-19T01:27:55.8859080Z -- Performing Test MM_MALLOC
2020-02-19T01:27:57.0625278Z -- Performing Test MM_MALLOC - Failed
2020-02-19T01:27:57.0759216Z -- Configuring done
2020-02-19T01:27:57.1151819Z -- Generating done
2020-02-19T01:27:57.1170218Z -- Build files have been written to: D:/a/1/s/build
2020-02-19T01:27:57.2783342Z Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
2020-02-19T01:27:57.2838140Z Copyright (C) Microsoft Corporation. All rights reserved.
2020-02-19T01:27:57.2875190Z 
2020-02-19T01:27:57.6015046Z   Checking Build System
2020-02-19T01:27:57.7328064Z   Building Custom Rule D:/a/1/s/CMakeLists.txt
2020-02-19T01:27:57.8770347Z   c_api.cpp
2020-02-19T01:27:57.8773171Z   lightgbm_R.cpp
2020-02-19T01:27:59.9723409Z   application.cpp
2020-02-19T01:28:00.9941809Z   boosting.cpp
2020-02-19T01:28:01.8875454Z   gbdt.cpp
2020-02-19T01:28:02.9367871Z   gbdt_model_text.cpp
2020-02-19T01:28:03.4741928Z   gbdt_prediction.cpp
2020-02-19T01:28:09.2936304Z   prediction_early_stop.cpp
2020-02-19T01:28:09.4802035Z   bin.cpp
2020-02-19T01:28:09.6131432Z   config.cpp
2020-02-19T01:28:09.6132016Z   config_auto.cpp
2020-02-19T01:28:09.6137385Z   dataset.cpp
2020-02-19T01:28:09.6137679Z   dataset_loader.cpp
2020-02-19T01:28:09.6832853Z   file_io.cpp
2020-02-19T01:28:09.6833465Z   json11.cpp
2020-02-19T01:28:09.6833701Z   metadata.cpp
2020-02-19T01:28:09.6833927Z   parser.cpp
2020-02-19T01:28:10.6142216Z   tree.cpp
2020-02-19T01:28:11.2481830Z   dcg_calculator.cpp
2020-02-19T01:28:12.1389082Z   metric.cpp
2020-02-19T01:28:12.8468733Z   linker_topo.cpp
2020-02-19T01:28:13.5626529Z   linkers_mpi.cpp
2020-02-19T01:28:13.6609369Z   linkers_socket.cpp
2020-02-19T01:28:13.6614767Z   network.cpp
2020-02-19T01:28:17.7698174Z   objective_function.cpp
2020-02-19T01:28:18.1152532Z   data_parallel_tree_learner.cpp
2020-02-19T01:28:19.3322662Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:19.3348768Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp)
2020-02-19T01:28:19.3356250Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp)
2020-02-19T01:28:19.4203533Z   feature_parallel_tree_learner.cpp
2020-02-19T01:28:19.4211880Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:19.4212828Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp)
2020-02-19T01:28:19.9075545Z   gpu_tree_learner.cpp
2020-02-19T01:28:19.9078201Z   serial_tree_learner.cpp
2020-02-19T01:28:20.8387353Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:20.8391501Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp)
2020-02-19T01:28:20.8401991Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp)
2020-02-19T01:28:20.8405676Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:20.8534636Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp)
2020-02-19T01:28:21.2776909Z   tree_learner.cpp
2020-02-19T01:28:21.4367192Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:21.4543088Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp)
2020-02-19T01:28:21.4550532Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp)
2020-02-19T01:28:21.4554342Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:21.7006770Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp)
2020-02-19T01:28:22.0023393Z   voting_parallel_tree_learner.cpp
2020-02-19T01:28:22.6860109Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:22.6860752Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp)
2020-02-19T01:28:22.8888505Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp)
2020-02-19T01:28:23.0199906Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:23.0234543Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp)
2020-02-19T01:28:23.3618217Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:23.3622647Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp)
2020-02-19T01:28:23.3625238Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp)
2020-02-19T01:28:23.3631018Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp) [D:\a\1\s\build\_lightgbm.vcxproj]
2020-02-19T01:28:23.3632364Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp)
2020-02-19T01:28:24.2276801Z   c_api.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
2020-02-19T01:28:25.7202949Z      Creating library D:/a/1/s/Release/lib_lightgbm.lib and object D:/a/1/s/Release/lib_lightgbm.exp
2020-02-19T01:28:25.7553411Z   Generating code
2020-02-19T01:28:53.7663906Z   Finished generating code
2020-02-19T01:28:54.5443573Z   _lightgbm.vcxproj -> D:\a\1\s\Release\lib_lightgbm.dll
2020-02-19T01:28:54.6154145Z   Building Custom Rule D:/a/1/s/CMakeLists.txt
2020-02-19T01:28:54.7546870Z   main.cpp
2020-02-19T01:28:54.7584845Z   application.cpp
2020-02-19T01:28:56.5499461Z   boosting.cpp
2020-02-19T01:28:56.8950578Z   gbdt.cpp
2020-02-19T01:28:58.2231404Z   gbdt_model_text.cpp
2020-02-19T01:28:58.4262242Z   gbdt_prediction.cpp
2020-02-19T01:28:59.8010973Z   prediction_early_stop.cpp
2020-02-19T01:29:00.0694204Z   bin.cpp
2020-02-19T01:29:00.3385029Z   config.cpp
2020-02-19T01:29:01.5289454Z   config_auto.cpp
2020-02-19T01:29:01.5319263Z   dataset.cpp
2020-02-19T01:29:03.0097908Z   dataset_loader.cpp
2020-02-19T01:29:03.3535710Z   file_io.cpp
2020-02-19T01:29:03.7173993Z   json11.cpp
2020-02-19T01:29:04.4702023Z   metadata.cpp
2020-02-19T01:29:04.8715889Z   parser.cpp
2020-02-19T01:29:06.0759989Z   tree.cpp
2020-02-19T01:29:06.2791605Z   dcg_calculator.cpp
2020-02-19T01:29:07.4230683Z   metric.cpp
2020-02-19T01:29:07.5752348Z   linker_topo.cpp
2020-02-19T01:29:08.5610320Z   linkers_mpi.cpp
2020-02-19T01:29:08.5640766Z   linkers_socket.cpp
2020-02-19T01:29:09.0953737Z   network.cpp
2020-02-19T01:29:10.6296354Z   objective_function.cpp
2020-02-19T01:29:10.9456559Z   data_parallel_tree_learner.cpp
2020-02-19T01:29:12.2380329Z   feature_parallel_tree_learner.cpp
2020-02-19T01:29:12.2765034Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:12.2767938Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp)
2020-02-19T01:29:12.2777307Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp)
2020-02-19T01:29:12.2783968Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:12.4758008Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\data_parallel_tree_learner.cpp)
2020-02-19T01:29:12.8528715Z   gpu_tree_learner.cpp
2020-02-19T01:29:12.9354165Z   serial_tree_learner.cpp
2020-02-19T01:29:13.5082377Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:13.7113036Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp)
2020-02-19T01:29:13.9279513Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp)
2020-02-19T01:29:13.9285254Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:13.9286170Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\feature_parallel_tree_learner.cpp)
2020-02-19T01:29:14.1175173Z   tree_learner.cpp
2020-02-19T01:29:14.3933261Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:14.3942440Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp)
2020-02-19T01:29:14.3950508Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp)
2020-02-19T01:29:14.3954262Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:14.5240935Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\serial_tree_learner.cpp)
2020-02-19T01:29:15.1126305Z   voting_parallel_tree_learner.cpp
2020-02-19T01:29:15.3523517Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:15.5411816Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp)
2020-02-19T01:29:15.5412597Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp)
2020-02-19T01:29:15.7504749Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:15.7506462Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\tree_learner.cpp)
2020-02-19T01:29:16.7765840Z d:\a\1\s\src\treelearner\feature_histogram.hpp(568): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:16.7766872Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(568): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp)
2020-02-19T01:29:16.7767260Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(109): note: see reference to function template instantiation 'void LightGBM::FeatureHistogram::FindBestThresholdSequence<false>(double,double,LightGBM::data_size_t,const LightGBM::ConstraintEntry &,double,LightGBM::SplitInfo *,int,bool,bool,int)' being compiled (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp)
2020-02-19T01:29:16.7767659Z d:\a\1\s\src\treelearner\feature_histogram.hpp(632): warning C4127: conditional expression is constant (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp) [D:\a\1\s\build\lightgbm.vcxproj]
2020-02-19T01:29:16.7767980Z   d:\a\1\s\src\treelearner\feature_histogram.hpp(632): note: consider using 'if constexpr' statement instead (compiling source file D:\a\1\s\src\treelearner\voting_parallel_tree_learner.cpp)
2020-02-19T01:29:17.3295773Z   main.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
2020-02-19T01:29:18.2528232Z      Creating library D:/a/1/s/Release/lightgbm.lib and object D:/a/1/s/Release/lightgbm.exp
2020-02-19T01:29:18.2882590Z   Generating code
2020-02-19T01:29:44.4845504Z   Finished generating code
2020-02-19T01:29:44.8725739Z   lightgbm.vcxproj -> D:\a\1\s\Release\lightgbm.exe
2020-02-19T01:29:44.9141703Z   Building Custom Rule D:/a/1/s/CMakeLists.txt
```

**Comment by guolinke** [comment]
these are removed in #2769

**Comment by StrikerRUS** [comment]
Great!

**Comment by StrikerRUS** [comment]
Fixed via #2769.
